File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
class Solution
{
    public int solution(int n, int a, int b)
    {
        int answer = 1;

        while(true) {
            if (Math.abs(a - b) == 1 && a / 2 != b / 2) {
                break;
            }
            
            a = a % 2 == 0 ? a / 2 : (a + 1) / 2;
            b = b % 2 == 0 ? b / 2 : (b + 1) / 2;
            answer++;
        }

        return answer;
    }
}

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.py
def solution(n,a,b):
    answer = 0
    width = 1
    
    while True:
        x = a-1
        y = b-1
        answer += 1
        width *= 2
        
        x = int(x/width)
        y = int(y/width)
        
        if x == y:
            break

    return answer